integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java

```diff
@@ -200,6 +200,7 @@ public void testCreateDistroXWithEncryptedVolumesInSingleRG(TestContext testCont
 
         testContext
                 .given(EnvironmentNetworkTestDto.class)
+                    .withServiceEndpoints()
```

Comment:
1. Based on our discussion this would be more robust as `.withServiceEndpoints(ServiceEndpointCreation.ENABLED_PRIVATE_ENDPOINT)`. So the related `EnvironmentNetworkTestDto` class also should be extended with:
```
...
    public EnvironmentNetworkTestDto valid() {
        return getCloudProvider()
                .network(this)
                .withServiceEndpoints();
    }
...
    public EnvironmentNetworkTestDto withServiceEndpoints(ServiceEndpointCreation serviceEndpointCreation) {
        getRequest().setServiceEndpointCreation(serviceEndpointCreation);
        return this;
    }
``` 
and the `withTryUseServiceEndpoints` should be removed from `EnvironmentNetworkTestDto`.
3. Could you create a separate validation for this new step and the related parameters? For example:
```
"serviceEndpointCreation": "ENABLED_PRIVATE_ENDPOINT",
"databasePrivateDnsZoneId": "/subscriptions/78fcb648.../resourceGroups/...dns-zone/providers/Microsoft.Network/privateDnsZones/privatelink.postgres.database.azure.com"
```
these should be present in the created environment response with correct values.

Reply:
Adding `.withServiceEndpoints`  to the EnvironmentNetworkTestDto would mean:
- Azure: use private endpoints if it is using single RG
- AWS: always use private endpoints

It means that for AWS we would not have tests without private endpoints at all.

Actually, when setting up a separate job for private endpoint tests I would prefer to specify the following parameters from environment variables:
- use private endpoints or not
- azure: use single RG or not

That is, decouple the use of single RG from the use of private endpoints.

Also, factoring creation / deletion of azure single RG out from the one distroX test that was set up with that, so that any test can be set up easily with an env parameter to be used with single RG.

However, I would leave that to a separate commit (this one is rather about adding a test for existing private DNS zone).

What do you think?